pytorch_lightning/utilities/hparams_mixin.py

```diff
@@ -0,0 +1,161 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import copy
+import inspect
+import re
+from argparse import Namespace
+from typing import Union, Any
+
+from pytorch_lightning.core.saving import PRIMITIVE_TYPES, ALLOWED_CONFIG_TYPES
+from pytorch_lightning.utilities import AttributeDict
+from pytorch_lightning.utilities.parsing import get_init_args
+
+
+class HyperparametersMixin:
```

Comment:
I am not very sure if we can introduce Mixins again, @williamFalcon

Reply:
@Borda Are there any updates on this?

Reply:
In general we shall avoid adding mixins which has some cross reference to other class attributes as they are uneasy to debug...
Cc: @tchaton